kubectl-plugin/src/resources/pool.rs

```diff
@@ -29,18 +50,41 @@ impl List for Pools {
 pub(crate) struct Pool {
     /// ID of the pool.
     id: PoolId,
+    output: OutputFormat,
 }
 
 #[async_trait(?Send)]
 impl Get for Pool {
     type ID = PoolId;
-    async fn get(id: &Self::ID) -> Result<()> {
-        let pool = RestClient::client()
-            .pools_api()
-            .get_pool(id)
-            .await
-            .map_err(|e| anyhow::anyhow!("Failed to get pool {}. Error {}", id, e))?;
-        println!("{:?}", pool);
-        Ok(())
+    type Format = OutputFormat;
+    async fn get(id: &Self::ID, output: &Self::Format) {
+        match RestClient::client().pools_api().get_pool(id).await {
+            Ok(pool) => match output.to_lowercase().trim() {
+                "" => {
+                    // Show the tabular form if outpur format is not specified.
+                    // Convert the output to a vector to be used in the method.
+                    let pool_to_vector: Vec<openapi::models::Pool> = vec![pool];
+                    let rows: Vec<Row> = utils::create_pool_rows(pool_to_vector);
+                    utils::table_printer((&*utils::POOLS_HEADERS).clone(), rows);
+                }
+                utils::YAML_FORMAT => {
+                    // Show the YAML form output if output format is YAML.
+                    let s = serde_yaml::to_string(&pool).unwrap();
+                    println!("{}", s);
+                }
+                utils::JSON_FORMAT => {
+                    // Show the JSON form output if output format is JSON.
+                    let s = serde_json::to_string(&pool).unwrap();
+                    println!("{}", s);
+                }
+                _ => {
```

Comment:
We could implement just one top-level check against provided output format and just use 100% correct value underneath.
A new enum can be introduced, which you can instantiate via 'try_from()' and then use directly in all your functions.

Reply:
Added a root check for the same. This would prevent an API call in case an output format is wrong.